de.bund.bfr.knime.fsklab.nodes/src/de/bund/bfr/knime/fsklab/nodes/ScriptHandler.java

```diff
@@ -103,8 +104,18 @@ public final void runSnippet(final FskPortObject fskObj, final FskSimulation sim
       
 
     exec.setProgress(0.75, "Run models script");
-    runScript(fskObj.getModel(), exec, false);
-
+    try {
+      runScript(fskObj.getModel(), exec, false);

```

Comment:
The catch only captures RException and IOException but runScript also throws CanceledExecutionException, InterruptedException, REXPMismatchException. The latter 3 are not subclasses of RException or IOException so they are not caught. *runSnippet* throws Exception so if one of the latter 3 exceptions occur they are leaked by runSnippet.

Reply:
@miguelalba runscript() with "false" argument never throws a REXPMismatchException. And I don't know if we should catch CanceledExecution or InterruptedException here.